### PR TITLE
Force cert reload on a daily basis.

### PR DESCRIPTION
**Pull request by itachiunited**
Forcing SSL reload once a day using a scheduled thread. This is to prevent any discrepancy due to FileWatcher in reloading the certs.

There has been scenarios where the FileWatcher is not reliable in detecting changes and reloading the certs as expected when changed. The certs were reloaded successfully in few components but it was never detected and run in others. This will result in few components with stale certificates. In order to prevent this issue, we are adding a new scheduled thread which will run once a day and reload the certs.

### CODE REVIEW

**Review by soumitra-st — APPROVED** [review]
:shipit:

File: pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java [hunk 1]
```diff
@@ -243,38 +264,52 @@ static void reloadSslFactoryWhenFileStoreChanges(SSLFactory baseSslFactory,
...
-          }
+
+          reloadSslFactory(baseSslFactory, keyStoreType, keyStorePath, keyStorePassword, trustStoreType, trustStorePath, trustStorePassword, sslContextProtocol, secureRandom, insecureModeSupplier);
         }
       }
       key.reset();
     }
   }
 
+  @VisibleForTesting
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java:275]: (minor) Should this be private?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java [hunk 2]
```diff
@@ -286,4 +321,4 @@ static void registerFile(WatchService watchService, Map<WatchKey, Set<Path>> key
     keyPathMap.computeIfAbsent(key, k -> new HashSet<>());
     keyPathMap.get(key).add(path.getFileName());
   }
-}
+}
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java:324]: (minor) Revert this

**Review by Jackie-Jiang — COMMENTED** [review]
High level question: Is it watcher based triggering right now? Seems `reloadSslFactoryWhenFileStoreChanges()` is also executed in a separate executor, is it possible to have a single entry point which monitors both triggering condition and interval?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java (hunk 2 above)
Comment (itachiunited) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java]: Done

File: pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java (hunk 1 above)
Comment (itachiunited) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/tls/RenewableTlsUtils.java]: Done.

**Comment by itachiunited** [comment]
> High level question: Is it watcher based triggering right now? Seems `reloadSslFactoryWhenFileStoreChanges()` is also executed in a separate executor, is it possible to have a single entry point which monitors both triggering condition and interval?

@Jackie-Jiang - Yes, the current flow is watcher based. We want to keep both separate since this is a backup to make sure that the certs are reloaded everyday. FileWatcher thread just has to run once after restart to set up the FileWatcher and the new executor is to make sure that it runs once a day to reload the SSL.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**